api/envoy/extensions/filters/http/ext_authz/v3/ext_authz.proto

```diff
@@ -63,6 +63,16 @@ message ExtAuthz {
   // <config_http_filters_ext_authz_stats>`.
   bool failure_mode_allow = 2;
 
+  //  This changes the filter's behaviour on errors:
+  //
+  //  1. When set to true, the filter would retain the status code received from the upstream and
+  //  would also propagate the response body and headers back to the client (if available) for
+  //  further debugging.
+  //
+  //  2. When set to false, the filter would return a 403 or `status_on_error` (if set) and would
+  //  not propagate any response body or headers back to the client. This is the default behavior.
+  bool allow_debugging_failures = 18;
```

Comment:
I think I would name this `propagate_response_on_failure` ?

Also, is this feature HTTP specific as it seems to pivot on 5xx? Should this actually be part of the HttpService object vs. global?

Reply:
What about adding a field analogous to `allowed_headers` for `allowed_response_headers`?

Reply:
> Also, is this feature HTTP specific as it seems to pivot on 5xx? Should this actually be part of the HttpService object vs. global?

Sounds reasonable.

Reply:
Thanks, @mattklein123. That sounds reasonable to me as well.

@ggreenway Sorry, I'm not able to understand the intention behind adding `allowed_response_header`. Is the idea to limit what could be propagated back to the client when there are failures? We expose [two lists](https://github.com/envoyproxy/envoy/blob/e802d9ee9ac2c355255b37afcf5c85e3e1fc82ee/api/envoy/extensions/filters/http/ext_authz/v3/ext_authz.proto#L305) `allowed_client_headers` and `allowed_upstream_headers` today which serves similar purpose.